Issue 6 add logger.

elipalmer opened the pull request:
Added basic logger initialization and configuration via the config/config file.  Tested by unit test and manual testing by changing the log_location config values, starting the studio, and verifying the file was created and written to.

I did not change existing console writes, as they are happening in scripting code and I am unsure of any effects this may have on those files.

[Issue #6 ](https://github.com/HarperDB/HarperDB_Studio/issues/6)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/harperdb/harperdb_studio/22)
<!-- Reviewable:end -->

Ethan-Arrowood commented on hdb_studio.js:
```diff
@@ -118,16 +114,20 @@ passport.deserializeUser(function (user, done) {
 
-function runServer() {
+async function runServer() {
+    await log_utils.initLogger().catch((e) => {
```
Wrong use of `await`.

Ethan-Arrowood replied:
Wrong use of `await`.
```javascript
try {
  await log_utils.initLogger();
} catch (e) {
  console.error(`...`);
}
```
or remove async await and just use `.catch`

Ethan-Arrowood commented on utility/logUtils.js:
```diff
@@ -0,0 +1,72 @@
+	}
+
+	if(config && config.log_location) {
+		let valid_path = await verifyLogLocation(config.log_location).catch( (e) => {
```
wrap this in a `try { } catch (e) { }` block

elipalmer commented:
*[utility/logUtils.js, line 54 at r1](https://reviewable.io/reviews/harperdb/harperdb_studio/22#-LELTbMAXCz94faZnirM-r1-54:-LEQQmD8BZofjJWzzOcf:b2nreh6) ([raw file](https://github.com/harperdb/harperdb_studio/blob/b2ed8f828ec2f3a1feef9eeeb60d6c197f73505b/utility/logUtils.js#L54)):*
<details><summary><i>Previously, Ethan-Arrowood (Ethan Arrowood) wrote…</i></summary><blockquote>

wrap this in a `try { } catch (e) { }` block
</blockquote></details>

Same as above.

---


*Comments from [Reviewable](https://reviewable.io/reviews/harperdb/harperdb_studio/22)*
<!-- Sent from Reviewable.io -->

Ethan-Arrowood commented on utility/logUtils.js:
```diff
@@ -51,9 +51,13 @@ async function initLogger() {
 
 	if(config && config.log_location) {
-		let valid_path = await verifyLogLocation(config.log_location).catch( (e) => {
+		let valid_path = undefined;
```
can be simplified to `let valid_path;`

elipalmer replied:
I need to define it outside of the try/catch block so it is properly scoped.

Ethan-Arrowood commented:
I'm running this locally and i'm getting an error:
`error getting stats for path ./log Error: ENOENT: no such file or directory, stat './log'`

Should the user have to initialize the log file manually or should the logger auto generate it? (imo the script should initialize the file if it doesn't exist)

elipalmer commented:
That error is due to the log directory not existing.  I added better verbiage describing a missing directory as well as added a little code to try to create the default log directory.  I also apparently just didn't add a unit test for initLogger() which was dumb, so I added some tests for it.

Tested by running the studio after removing the log directory, leaving the existing log directory, and creating a custom path in the config file.  All unit tests pass.

elipalmer commented:
Updated with a merge from master to resolve conflicts.  @Ethan-Arrowood or @Alyssa-Cooper could you take another look at this?

Ethan-Arrowood commented on utility/logUtils.js:
```diff
@@ -0,0 +1,103 @@
+ */
+async function verifyLogLocation(log_path) {
+  if (!log_path) {
+    return false;
```
nit: one line code blocks should be inlined:
